modules/packages/Python.chpl

```diff
@@ -2217,6 +2217,157 @@ module Python {
     proc getPyObject() do return this.obj;
 
     // TODO: call should support kwargs
+
+    // Casts
+    /* Creates a new int from ``x``, when ``x`` is a :class:`Value`. */
+    operator :(x: borrowed Value, type t: int(?)): t throws {
+      var g = PyGILState_Ensure();
+      defer PyGILState_Release(g);
+
+      var pyObj = x.getPyObject();
+      Py_INCREF(pyObj);
+      var res = x.interpreter.fromPythonInner(t, pyObj);
+
+      return res;
+    }
+
+    /* Creates a new uint from ``x``, when ``x`` is a :class:`Value`. */
+    operator :(x: borrowed Value, type t: uint(?)): t throws {
+      var g = PyGILState_Ensure();
+      defer PyGILState_Release(g);
+
+      var pyObj = x.getPyObject();
+      Py_INCREF(pyObj);
+      var res = x.interpreter.fromPythonInner(t, pyObj);
+
+      return res;
+    }
+
+    /* Creates a new real from ``x``, when ``x`` is a :class:`Value`. */
+    operator :(x: borrowed Value, type t: real(?)): t throws {
+      var g = PyGILState_Ensure();
+      defer PyGILState_Release(g);
+
+      var pyObj = x.getPyObject();
+      Py_INCREF(pyObj);
+      var res = x.interpreter.fromPythonInner(t, pyObj);
+
+      return res;
+    }
+
+    /* Creates a new bool from ``x``, when ``x`` is a :class:`Value`. */
+    operator :(x: borrowed Value, type t: bool): t throws {
+      var g = PyGILState_Ensure();
+      defer PyGILState_Release(g);
+
+      var pyObj = x.getPyObject();
+      Py_INCREF(pyObj);
+      var res = x.interpreter.fromPythonInner(t, pyObj);
+
+      return res;
+    }
+
+    /* Creates a new string from ``x``, when ``x`` is a :class:`Value`. */
+    operator :(x: borrowed Value, type t: string): t throws {
+      var g = PyGILState_Ensure();
```

Comment:
Does this function not conflict with the existing cast to string for Value objects?

I am also not sure why you need special logic here, the existing cast should do the right thing for both strings and nom-strings

Reply:
Sigh.  I did look for an already existing operator, but it looks like I was searching for `operator :` and so missed `operator:`.  I'll remove this one

Reply:
I think you should also consider expanding `test/library/packages/Python/correctness/objToString.chpl` with a cast to string